Set CMAKE_TOOLCHAIN_FILE for nanoflann build

Otherwise, the CMakeTestCCompiler.cmake test will fail.  The additional compiler options
will be lost when invoking the compiler driver, resulting in the linker failing to
find the crt libraries.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

diff --git a/cmakemodules/script_nanoflann.cmake b/cmakemodules/script_nanoflann.cmake
index 82a07cae..98078469 100644
--- a/cmakemodules/script_nanoflann.cmake
+++ b/cmakemodules/script_nanoflann.cmake
@@ -19,6 +19,7 @@ if (NOT nanoflann_FOUND)
         -DNANOFLANN_BUILD_EXAMPLES=OFF
         -DNANOFLANN_BUILD_TESTS=OFF
         -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+	-DCMAKE_TOOLCHAIN_FILE=${PROJECT_SOURCE_DIR}/../toolchain.cmake
       RESULT_VARIABLE result
       WORKING_DIRECTORY "${nanoflann_EMBEDDED_BUILD_DIR}"
       ${echo_flag}
